rest/app.py

```diff
@@ -1,107 +1,105 @@
+sql_users = SQLUsers()
+sql_tasks = SQLTasks()
 
 
 @app.route('/')
 @app.route('/index')
 def index():
-    return render_template('index.html', users=users, tasks=tasks)
+    return render_template('index.html')
 
 
 @app.route('/users', methods=['GET'])
 def get_users():
-    return json.dumps(users)
+    return json.dumps(sql_users.get_all_users())
 
 
 @app.route('/users', methods=['POST'])
 def add_user():
     user = json.loads(request.json)
-    users.append(user)
-    return render_template('index.html'), 201
+    user_name = user['name']
+    added_user = sql_users.add_user(user_name)
+    return json.dumps(added_user), 201
 
 
 @app.route('/users/<int:user_id>', methods=['DELETE'])
 def delete_user(user_id):
-    for user in users:
-        if user['id'] == user_id:
-            users.remove(user)
-            return render_template('index.html'), 201
-    return not_found(404)
+    sql_users.delete_user(user_id)
+    return render_template('index.html'), 204
 
 
 @app.route('/users/<int:user_id>', methods=['GET'])
 def get_user(user_id):
-    for user in users:
-        if user['id'] == user_id:
-            return json.dumps(user)
-    return not_found(404)
+    user = sql_users.get_user(user_id)
+    return json.dumps(user) if user else not_found(404)
 
 
 @app.route('/users/<int:user_id>/tasks', methods=['GET'])
-def get_user_tasks(user_id, start='', end=''):
-    # ToDo(stasya): add request to database
-    user_tasks = list()
-    for task in tasks:
-        if task['user_id'] == user_id:
-            # SQL request for the given period
-            user_tasks.append(task)
-
-    return json.dumps(user_tasks)
+def get_user_tasks(user_id):
+    """
+    Get tasks for user with specified ID.
+    Request args may be:
+    empty - returns all user tasks
+    period (in format year, [month], [day]) - return user tasks in accordance
+    with this criteria
+
+    :return: list of user tasks for the given period or all of the tasks
+    """
+    period = request.args
+    if period:
+        year = period['year']
+        month = period.get('month', None)
+        day = period.get('day', None)
+        user_tasks = sql_tasks.get_tasks_for_period(user_id, year, month, day)
+    else:
+        user_tasks = sql_tasks.get_all_user_tasks(user_id)
+    return json.dumps(user_tasks, default=str)
 
 
 @app.route('/tasks/<int:task_id>', methods=['GET'])
-def get_task(task_id):
-    for task in tasks:
-        if task['id'] == task_id:
-            return json.dumps(task)
-    return not_found(404)
+def get_task_by_id(task_id):
+    task = sql_tasks.get_task(task_id)
+    return json.dumps(task) if task else not_found(404)
 
 
 @app.route('/tasks', methods=['POST'])
 def add_task():
     # ToDo(stasya) Add json validation
     # https://github.com/stasya72008/highway-to-hell/projects/2#card-11155844
     task = json.loads(request.json)
-    tasks.append(task)
-    return render_template('index.html'), 201
+    user_id = task['user_id']

```

Comment:
I think You can drop lines _user_id = task['user-id']_
and send this parameter directly  
sql_tasks.add_task(task['user-id'], ...)